benchmarks/src/main/java/org/apache/druid/benchmark/query/ScanBenchmark.java

```diff
@@ -0,0 +1,404 @@
+    final BenchmarkSchemaInfo basicSchema = BenchmarkSchemas.SCHEMA_MAP.get("basic");
+
+    final List<String> queryTypes = ImmutableList.of("A", "B", "C", "D");
+    for (final String eachType : queryTypes) {
+      basicQueries.put(eachType, makeQuery(eachType, basicSchema));
+    }
+
+    SCHEMA_QUERY_MAP.put("basic", basicQueries);
+  }
+
+  private static Druids.ScanQueryBuilder makeQuery(final String name, final BenchmarkSchemaInfo basicSchema)
+  {
+    switch (name) {
+      case "A":
+        return basicA(basicSchema);
+      case "B":
+        return basicB(basicSchema);
+      case "C":
+        return basicC(basicSchema);
+      case "D":
+        return basicD(basicSchema);
+      default:
+        return null;
+    }
+  }
+
+  /* Just get everything */
+  private static Druids.ScanQueryBuilder basicA(final BenchmarkSchemaInfo basicSchema)
+  {
+    final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Collections.singletonList(basicSchema.getDataInterval()));
+
+    return Druids.newScanQueryBuilder()
+                 .dataSource("blah")
+                 .intervals(intervalSpec);
+
+  }
+
+  private static Druids.ScanQueryBuilder basicB(final BenchmarkSchemaInfo basicSchema)
+  {
+    final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Collections.singletonList(basicSchema.getDataInterval()));
+
+    final List<String> dimUniformFilterVals = new ArrayList<>();
+    int resultNum = (int) (100000 * 0.1);
+    int step = 100000 / resultNum;
+    for (int i = 1; i < 100001 && dimUniformFilterVals.size() < resultNum; i += step) {
+      dimUniformFilterVals.add(String.valueOf(i));
+    }
+
+    List<String> dimHyperUniqueFilterVals = new ArrayList<>();
+    resultNum = (int) (100000 * 0.1);
+    step = 100000 / resultNum;
+    for (int i = 0; i < 100001 && dimHyperUniqueFilterVals.size() < resultNum; i += step) {
+      dimHyperUniqueFilterVals.add(String.valueOf(i));
+    }
+
+    DimFilter filter = new InDimFilter("dimHyperUnique", dimHyperUniqueFilterVals, null);
+
+    return Druids.newScanQueryBuilder()
+                 .filters(filter)
+                 .intervals(intervalSpec);
+  }
+
+  private static Druids.ScanQueryBuilder basicC(final BenchmarkSchemaInfo basicSchema)
+  {
+    final QuerySegmentSpec intervalSpec = new MultipleIntervalSegmentSpec(Collections.singletonList(basicSchema.getDataInterval()));
+
+    final List<String> dimUniformFilterVals = new ArrayList<>();
+    final int resultNum = (int) (100000 * 0.1);
+    final int step = 100000 / resultNum;
+    for (int i = 1; i < 100001 && dimUniformFilterVals.size() < resultNum; i += step) {
+      dimUniformFilterVals.add(String.valueOf(i));
+    }
+
+    final String dimName = "dimUniform";
+    return Druids.newScanQueryBuilder()
+        .filters(new SelectorDimFilter(dimName, "3", StrlenExtractionFn.instance()))
+        .intervals(intervalSpec); // TODO
```

Comment:
What's the TODO here?

Reply:
Oops didn't commit the changes where I removed that.